packages/dd-trace/src/llmobs/plugins/genai.js

```diff
@@ -0,0 +1,518 @@
+    })
+  }
+
+  // ============================================================================
+  // Public API Methods
+  // ============================================================================
+
+  getLLMObsSpanRegisterOptions (ctx) {
+    const methodName = ctx.methodName
+    if (!methodName) return
+
+    const inputs = ctx.inputs
+    const operation = getOperation(methodName)
+    const kind = operation === 'embedding' ? 'embedding' : 'llm'
+
+    return {
+      modelProvider: 'google',
+      modelName: inputs.model,
+      kind,
+      name: 'google_genai.request'
+    }
+  }
+
+  setLLMObsTags (ctx) {
+    const span = ctx.currentStore?.span
+    const methodName = ctx.methodName
+    if (!methodName) return
+
+    const inputs = ctx.inputs
+    const response = ctx.result
+    const error = !!span.context()._tags.error
+
+    const operation = getOperation(methodName)
+
+    if (operation === 'chat') {
+      this._tagGenerateContent(span, inputs, response, error, ctx.isStreaming)
+    } else if (operation === 'embedding') {
+      this._tagEmbedding(span, inputs, response, error)
+    }
+
+    if (!error && response) {
+      const metrics = this._extractMetrics(response)
+      this._tagger.tagMetrics(span, metrics)
+    }
+  }
+
+  // ============================================================================
+  // Streaming Utilities
+  // ============================================================================
+
+  _aggregateStreamingChunks (chunks) {
+    const response = { candidates: [] }
+
+    for (const chunk of chunks) {
+      if (chunk.candidates) {
+        // Flatten candidates array
+        response.candidates.push(...chunk.candidates)
+      }
+      if (chunk.usageMetadata) {
+        response.usageMetadata = chunk.usageMetadata
+      }
+    }
+
+    return response
+  }
+
+  // ============================================================================
+  // Tagging Methods
+  // ============================================================================
+
+  _tagGenerateContent (span, inputs, response, error, isStreaming = false) {
+    const { config = {} } = inputs
+
+    const inputMessages = this._formatInputMessages(inputs.contents)
+
+    if (error) {
+      this._tagger.tagLLMIO(span, inputMessages, [{ content: '' }])
+      return
+    }
+
+    const outputMessages = this._formatOutputMessages(response, isStreaming)
+    this._tagger.tagLLMIO(span, inputMessages, outputMessages)
+
+    const metadata = this._extractMetadata(config)
+    this._tagger.tagMetadata(span, metadata)
+  }
+
+  _tagEmbedding (span, inputs, response, error) {
+    const embeddingInput = this._formatEmbeddingInput(inputs.contents)
+
+    if (error) {
+      this._tagger.tagEmbeddingIO(span, embeddingInput)
+      return
+    }
+
+    const embeddingOutput = this._formatEmbeddingOutput(response)
+    this._tagger.tagEmbeddingIO(span, embeddingInput, embeddingOutput)
+    this._tagger.tagMetadata(span, {})
```

Comment:
we can get rid of this call i think, since it's setting an empty metadata object (`{}`)